source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1195,6 +1196,56 @@ namespace
         }
     }
 
+    void writeMasterPrim(DocumentPtr document, RtStagePtr rtStage, const RtPrim& masterPrim)
+    {
+        if (!masterPrim.isValid())
+        {
+            return;
+        }
+
+        // Write the definition
+        PvtStage* stage = PvtStage::ptr(rtStage);
+        PvtPrim* prim = PvtObject::ptr<PvtPrim>(masterPrim);
+        writeNodeDef(prim, document);
+
+        // Write the corresponding nodegraph if any
+        RtNodeDef def(masterPrim);
+        RtToken nodeDefName = def.getName();
+        RtSchemaPredicate<RtNodeGraph> filter;
+        for (RtPrim child : stage->getRootPrim()->getChildren(filter))
```

Comment:
Doesn't the rtStage have a getRootPrim method? Why do you need the private stage?

Reply:
Note: we are using a public iterator interface returned from getChildren().